crates/sui-core/src/safe_client.rs

```diff
@@ -210,48 +210,45 @@ impl<C> SafeClient<C> {
     fn check_transaction_info(
         &self,
         digest: &TransactionDigest,
-        response: TransactionInfoResponse,
+        transaction: VerifiedTransaction,
+        status: TransactionStatus,
     ) -> SuiResult<VerifiedTransactionInfoResponse> {
-        match response {
-            TransactionInfoResponse::Signed(signed) => {
-                fp_ensure!(
-                    digest == signed.digest(),
-                    SuiError::ByzantineAuthoritySuspicion {
-                        authority: self.address,
-                        reason: "Signed transaction digest does not match with expected digest"
-                            .to_string()
-                    }
-                );
-                let committee = self.get_committee(&signed.epoch())?;
-                Ok(VerifiedTransactionInfoResponse::Signed(
-                    signed.verify(&committee)?,
-                ))
-            }
-            TransactionInfoResponse::Executed(cert, effects) => {
-                Ok(VerifiedTransactionInfoResponse::Executed(
-                    self.check_certificate(cert, digest)?,
-                    self.check_signed_effects(digest, effects, None)?,
-                ))
-            }
-        }
-    }
-
-    fn check_certificate(
-        &self,
-        certificate: CertifiedTransaction,
-        expected_digest: &TransactionDigest,
-    ) -> SuiResult<VerifiedCertificate> {
-        // Check it's the right transaction
         fp_ensure!(
-            certificate.digest() == expected_digest,
+            digest == transaction.digest(),
```

Comment:
in the write path the `digest` is a reference to `transaction`, see line 291

Reply:
But not in the read path. So we need this code somewhere.

Reply:
we could move that fp_ensure out to the read path callsite.

None of these are blocking comments anyway, feel free to land.